server/channels/app/imports/import_validators.go

```diff
@@ -650,3 +654,37 @@ func isValidEmailBatchingInterval(emailInterval string) bool {
 		emailInterval == model.PreferenceEmailIntervalFifteen ||
 		emailInterval == model.PreferenceEmailIntervalHour
 }
+
+// isValidGuestRoles checks if the user has both guest roles in the same team or channel.
+// at this point we assume that the user has a valid role scheme.
+func isValidGuestRoles(data UserImportData) bool {
+	var isSystemGuest bool
+	if data.Roles != nil {
```

Comment:
This check is already done above.

Reply:
Right, maybe we should avoid that check in the upper level, seems like we are okay with empty roles in the UserImportData afterall.